src/helpers.rs

```diff
@@ -827,6 +881,123 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
             None => tcx.item_name(def_id),
         }
     }
+
+    // extract the scalar value from the result of reading a scalar from the machine
+    // and convert it to a CArg
```

Comment:
General notes on documentation and comments:

* Prefer doc comments over regular comments (if the site allows doc comments, which this one does)
* Start with a capitalized letter
* Put things like `CArg` or other names into backticks
* End sentences with a `.`